src/components/data-table/column-sorting/column-sort_spec.ts

```diff
@@ -0,0 +1,203 @@
+import {Component, DebugElement, ViewChildren, QueryList, OnInit} from '@angular/core';
+import {CORE_DIRECTIVES} from '@angular/common';
+import {By} from "@angular/platform-browser";
+import {
+  MdDataColumnSortingService,
+  MdDataColumnSortDirective,
+  SortDirection,
+  IColumnSortingModel
+} from './index';
+
+export function main() {
+
+  @Component({
+    selector: 'colsort-test-cmp',
+    providers: [MdDataColumnSortingService],
+    directives: [CORE_DIRECTIVES, MdDataColumnSortDirective],
+    template: `
+    <table>
+      <thead>
+        <tr>
+          <th md-data-column-sort="1">Wine</th>
+          <th md-data-column-sort="2">Cheese</th>
+          <th md-data-column-sort="3">Crackers</th>
+        </tr>
+      </thead>
+    </table>`
+  })
+  class ColsortTestComponent implements OnInit {
+    defaultModel: IColumnSortingModel = {
+      column: '1',
+      direction: SortDirection.DESCEND
+    };
+    @ViewChildren(MdDataColumnSortDirective) columns: QueryList<MdDataColumnSortDirective>;
+
+    /**
+     * Constructor.
+     * @param sortingService - reference to the service for this area.
+     */
+    constructor(public sortingService: MdDataColumnSortingService) { }
+
+    ngOnInit() {
+      this.sortingService.setSorting(this.defaultModel);
+    }
+  }
+
+  let fixture: ComponentFixture<ColsortTestComponent>;
+  let service: MdDataColumnSortingService;
+  let testCmp: ColsortTestComponent;
+
+  beforeEachProviders(() => [
+    TestComponentBuilder
+  ]);
+
+  beforeEach(injectAsync([TestComponentBuilder], (tcb) => {
+    return tcb
+      .createAsync(ColsortTestComponent)
+      .then(f => {
+        f.detectChanges();
+        fixture = f;
+        testCmp = fixture.componentInstance;
+      })
+      .catch(console.error.bind(console));
+  }));
+
+  describe('MdDataSortColumnService', () => {
+    describe('sortingColumn', () => {
+
+      it('is observable sorting state', () => {
+        expect(testCmp.sortingService.sortingColumn$.subscribe).toBeDefined();
+      });
+
+    });
+
+    describe('setSorting', () => {
+      it('sets sorting programatically', done => {
+        let testSort = { column: '1', direction: SortDirection.ASCEND };
+        expect(testCmp.sortingService.sortingColumn$).toBeDefined();
+
+        testCmp.sortingService.sortingColumn$.subscribe(col => {
+          expect(col).toEqual(testSort);
+          done();
+        });
+
+        testCmp.sortingService.setSorting(testSort);
+      });
+    });
+
+    describe('changeSorting', () => {
+      it('updates sorting based on column interaction', done => {
+        let clicks = 0;
+        let currentSort = { column: '1', direction: SortDirection.DESCEND };
+        let firstSort = { column: '3', direction: SortDirection.ASCEND };
+        let secondSort = { column: '3', direction: SortDirection.DESCEND };
+        let thirdSort = firstSort;
+
+        testCmp.sortingService.sortingColumn$.subscribe(col => {
```

Comment:
Added second it case. I'm going to wait for @justindujardin to comment about test format before doing any refactoring in that regard.